.github/workflows/build.yml

```diff
@@ -0,0 +1,25 @@
+name: build
+
+on:
+  push:
+    branches:
+      - main
+      - release/**
+
+jobs:
+  dist:
+    name: Build wheel and .tar files for release
+    timeout-minutes: 10
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: 3.12
```

Comment:
just a nit but since the repo we are using it in is python 3.11 i'd probably run this in a matrix against multiple python versions